File: nim/sieve/sieve.nim
import sequtils

func primes*(maxPrime: int): seq[int] =
  if maxPrime < 2:
    return @[]
  var
    candidates = (2..maxPrime).toSeq
    marks = newSeq[bool](candidates.len)
  for candidate in candidates:
    var i = 2 * candidate - 2
    while i < maxPrime - 1:
      marks[i] = true
      i += candidate
  for i in marks.low..marks.high:
    if not marks[i]:
      result.add candidates[i]

File: python/sieve/sieve.py
from typing import List


def primes(limit: int) -> List[int]:
    ret = list()
    if limit >= 2:
        candidates = list(range(2, limit + 1))
        marks = [False for _ in range(len(candidates))]
        for candidate in candidates:
            i = 2 * candidate - 2
            while i < limit - 1:
                marks[i] = True
                i += candidate
        for i, mark in enumerate(marks):
            if not mark:
                ret.append(candidates[i])
    return ret
